olive/engine/cloud_cache_helper.py

```diff
@@ -92,22 +99,20 @@ def get_hash_key(
         self, model_config: ModelConfig, pass_search_point: Dict[str, Any], input_model_hash: Optional[str]
     ):
         """Get hash key from input model config, pass search point, and input model hash."""
-        try:
-            from huggingface_hub import repo_info
-        except ImportError:
-            logger.exception(
-                "huggingface_hub is not installed. "
-                "Please install huggingface_hub to use the cloud model cache feature for Huggingface model."
-            )
-
         hf_hub_model_commit_id = None
         model_config_copy = deepcopy(model_config)
-        if (
-            input_model_hash is None
-            and model_config.type.lower() == "hfmodel"
-            and create_resource_path(model_config.config["model_path"]).is_string_name()
-        ):
-            hf_hub_model_commit_id = repo_info(model_config.config["model_path"]).sha
+        model_path_resource_path = create_resource_path(model_config.config.get("model_path"))
+        if input_model_hash is None:
```

Comment:
following the logic in engine `input_model_hash` is always called with `none` so the else part is never run. Is there a reason for this?

Reply:
Only the first input model hash for first Pass is None. The following pass's input model hash will be the output model hash from previous pass run returns in the for loop.